tests/test_config_command.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use regex::Regex;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+#[test]
+fn test_config_get() {
+    let test_env = TestEnvironment::default();
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "get", "user.name"]);
+    insta::assert_snapshot!(stdout, @"Test User");
+}
```

Comment:
Could you add a test showing the output for a table (e.g. `jj config get user`)? FYI, `hg config` shows just the value if it's not a table and it shows the full path for each value if it is a table. That seems inconsistent but pragmatic, so we may want to copy that. For example:

```
$ hg config diff
diff.git=1
diff.showfunc=1
diff.word-diff=1
diff.merge=true
$ hg config diff.merge
true
```

Actually, do we need `get` and `list` as separate subcommands or should we copy hg's behavior?

Reply:
> do we need `get` and `list` as separate subcommands or should we copy hg's behavior?

`get` and `list` can be merged (because they are immutable), but I think it's better to make `set` a subcommand or separate command: `config set`, `configset`, or `setconfig`.

Reply:
Oh, good idea! Then should it still have separate `get` and `list` commands? Should one be an alias of the other?

BTW, I made `get`/`list` an explicit subcommand instead of supporting `jj config somekey` like git does. WDYT about that? The alternatives all seemed messy (overloaded positional args, using flags as subcommands like git's `--list`, or having an entirely separate command like `setconfig`).

Reply:
> Then should it still have separate `get` and `list` commands?

Probably not. I'd go with just `list`.

> BTW, I made `get`/`list` an explicit subcommand instead of supporting `jj config somekey` like git does.

I agree with that. I think it's good to separate readonly commands from mutating commands (even though "readonly" is a bit funny in a VCS that automatically snapshots the working copy). My preference for that command is `jj config set`.

Reply:
> I'd go with just `list`.
Hmm, I'd imagined `config get` and `config set` being parallel:
```
$ jj config get foo
someval
$ jj config set foo otherval
```

But if we just have `list` and that always prints the key name, it's kind of annoyingly unparallel:
```
$ jj config list foo
foo="someval"
$ jj config set foo otherval
```
Is that not a concern? git has them as separate commands. hg doesn't but doesn't seem to have any equivalent of `config set` either.

Reply:
> But if we just have `list` and that always prints the key name, it's kind of annoyingly unparallel:
> 
> ```
> $ jj config list foo
> foo="someval"
> $ jj config set foo otherval
> ```

See the first message in the thread where I suggested copying `hg`'s behavior of skipping the "foo=" bit for simple values. What do you think? I'm fine with still having separate `get` and `list` commands, but what would the difference be? Would `get` only be allowed/useful on single values (like `git config --get`)? Would `list` be able to restrict the subset by specifying a table name (e.g. `jj config list merge-tools`)?

Reply:
Right, `get` on single values, `list` for any value type. I suppose we can just have `list` for now and consider `get` later, at long as we get the behavior pretty sensible and future-proof.

I think I should make `list` always output valid TOML for consistency, though. I didn't get it quoting strings properly yet. Can do that before we merge if that SGTY.

Reply:
Oh and

> See the first message in the thread where I suggested copying hg's behavior of skipping the "foo=" bit for simple values.

Could do that. I kinda prefer predictable output format. How about we just do a simple `list` for now with pretty simple behavior and follow up on shortening output / other commands/ etc in future changes?

Reply:
SGTM

Reply:
> I think I should make `list` always output valid TOML for consistency

Good point. I had not even thought about that. That seems useful, not only to make it easy to copy and paste, but we kind of have to do that for list-valued entries.

Reply:
Think I'm close to having something working, but I'm having noob issues converting `config::Value`s into something `toml::to_string` will accept. Does it make sense to construct a cloned subset of the whole config and serialize that? Am I way overcomplicating this?

I could just special-case to quote strings and format arrays, but it feels like I oughta go ahead and deserialize properly. WDYT?

Reply:
Hmm, that's a good question. I suppose the problem stems from the fact that the `config` crate is designed to work with different formats, so we could hypothetically have config in files that are not TOML files, *and* the `config` crate is only designed for reading config. I'm fine with special-casing for now. By the way, I read that the `config` crate may gain support writing configs.

Reply:
Okay, special-cased for now. LMK if any of the code is especially atrocious, but it seems to work okay for interesting cases and always output valid TOML now.